packages/embark-solidity-tests/src/lib/index.js

```diff
@@ -34,53 +47,102 @@ class SolidityTestRunner {
   }
 
   match(path) {
     return SOLIDITY_TEST_MATCH.test(path);
   }
 
-  run() {
+  run(options, cb) {
+    const reporter = new Reporter(options.reporter);
     const {events} = this;
 
+    if (this.files.length === 0) {
+      return cb(null, 0);
+    }
+
     let web3;
-    let contractFiles = [];
-    let compiledContracts = [];
+    let contractsToDeploy = {};
 
-    this.files.forEach(f => {
-      const resolver = (cb) => {
-        fs.readFile(file, (err, data) => { cb(data.toString()); });
-      };
+    const resolverFn = (file) => {
+      return (cb) => {
+        fs.readFile(file, (_err, data) => {
+          const code = data.toString();
+          const injected = this.prepare(code);
 
-      const file = new File({
-        path: f,
-        originalPath: f,
-        type: Types.dappFile,
-        resolver
-      });
+          cb(injected);
+        });
+      };
+    };
 
-      contractFiles.add(file);
-    });
+    const contractFiles = this.files.map(f => new File({path: f, originalPath: f, type: Types.dappFile, resolver: resolverFn(f)}));
+    contractFiles.unshift(ASSERT_LIB);
 
     async.waterfall([
+      (next) => {
+        // write the remix_tests file where it will be found.
+        fs.writeFile(dappPath('remix_tests.sol'), remixTests.assertLibCode, next);
+      },
+      (next) => {
+        events.request("contracts:reset", next);
+      },
       (next) => {
         events.request("compiler:contracts:compile", contractFiles, next);
       },
-      (cc, next) => {
-        events.request("contracts:build", {}, cc, next);
+      (compiledContracts, next) => {
+        // TODO: fetch config and use it here
+        events.request("contracts:build", { contracts: {} }, compiledContracts, next);
       },
-      (contractsList, contractsDeps, next) => {
-        events.request("deployment:contracts:deploy", contractsList, contractDeps, next);
+      (cl, cd, next) => {
+        // Build the list of contracts we actually want to deploy
+        contractsToDeploy = { Assert: cl['Assert'] };
+
+        for(const contract in cl) {
+          if (!this.match(cl[contract].path)) {
+            continue;
+          }
+
+          cl[contract].gas = GAS_LIMIT;
+          contractsToDeploy[contract] = cl[contract];
+        }
+
+        events.request("deployment:contracts:deploy", contractsToDeploy, cd, next);
       },
-      (next) => {
+      (_foo, next) => {
         events.request("blockchain:client:provider", "ethereum", next);
       },
       (bcProvider, next) => {
         web3 = new Web3(bcProvider);
+        web3.eth.getAccounts(next);
       },
-
-    ]);
+      (accounts, next) => {
+        const opts = { from: accounts[0] };
+        const contracts = Object.keys(contractsToDeploy).filter(c => c !== "Assert");
+
+        async.eachOfLimit(contracts, 1, (contract, _, cb) => {
+          const instance = contractsToDeploy[contract];
+
+          const contractObj = new web3.eth.Contract(
```

Comment:
Instead of coupling this with web3, you can do `request2("blockchain:contract:create", { abi: instance.abiDefinition, address: instance.deployedAddress, from: accounts[0] });'`, though it's only available once https://github.com/embark-framework/embark/pull/1831 is merged.